subprojects/core/src/main/java/org/gradle/internal/exceptions/ContextAwareException.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.exceptions;
+
+import org.gradle.api.GradleException;
+import org.gradle.util.TreeVisitor;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ContextAwareException extends GradleException {
+
+    public ContextAwareException(Throwable t) {
+        initCause(t);
+    }
+
+    public String getLocation() {
+        return null;
+    }
```

Comment:
I do not like having this here only to be overriden in LocationAwareException. However, I don't like even more having to `instanceof` in `BuildExceptionReporter` to figure if the exception is location-aware.

I have ideas on how to improve this in the next iteration and would rather handle it in subsequent changes. One such change is [this](https://github.com/gradle/gradle/pull/12195).

Reply:
I'm thinking something like [this](https://github.com/gradle/gradle/pull/12200/files) next - capture the details using a visitor like we do for reportable causes.